doc/ipmc/IPMC_Data_Plane_HLD.md

```diff
@@ -0,0 +1,472 @@
+
+Schema:
+
+```SQL
+; New table
+; Store IP multicast routing data
+
+key = MROUTE_TABLE:vrf_name|source_ip|dest_ip 
+                                    ; APPL DB usually uses ':' as the separator, but '|' is chosen here.
+                                    ; The reason is that when the ip address is ipv6, using ':' cannot distinguish two ipv6 addresses
+                                    vrf_name    ; The multicast route belongs to the Vrf. If this field is empty, the default global Vrf is used
+                                    source_ip   ; Source IP addresses of multicast routes, for example, IPv4 0.0.0.0(ASM), A.B.C.D(SSM)
+                                    dest_ip     ; Indicates the destination IP address of the multicast route
+
+; field = value
+incoming_group = router_intfs_group  ; IP multicast routes into an interface member group, which can be empty or have multiple members. Each member is separated by commas ','
+outgoing_group = router_intfs_group  ; The IP multicast route outgoing interface member group, which cannot be empty. It can have one or more members. Each member is separated by commas ','
+```
+
+Example:
+
+```bash
+admin@sonic:~$ redis-cli -n 4 hgetall "MROUTE_TABLE|0.0.0.0|225.0.0.1"
+1) "incoming_group"
+2) "Ethernet1"
+3) "outgoing_group"
+4) "Ethernet13,Ethernet15"
+```
+
+```bash
+admin@sonic:~$ redis-cli -n 0 hgetall "MROUTE_TABLE:192.168.1.200|230.0.0.1"
+1) "incoming_group"
+2) "Ethernet49"
+3) "outgoing_group"
+4) "Ethernet39,Ethernet65,Vlan200"
+```
+
+#### SWSS Container
+
+mroutesyncd
+
+- Reads IP multicast routing messages from the kernel and listens for IP multicast routing changes in the kernel
+- Based on netlink messages, the source IP address, destination IP address, inbound interface member, and outbound member of the IP multicast route are resolved and written into the APPL DB
+- During a warm reboot, compare the kernel multicast route and APPL DB data to update the warm reboot data
+
+intfsorch
+
+- Listens to INTF_TABLE in the APPL DB and enables the IP multicast forwarding function of the corresponding routing interface
+
+mgmanager
+
+- IP Multicast route encapsulation management of incoming and outgoing interface groups
+- Encapsulate and manage the creation, modification, and deletion of rpf group and ipmc group
+- Implement rpf/ipmc group and member reference counting, exception handling, and other behaviors
+- Internally invoke the SAI API of the rpf/ipmc group and member. For details about the SAI API, see the following
+- Provides external query and update interfaces for the rpf/ipmc group
+
+mrouteorch
+
+- Listens to MROUTE_TABLE in the APPL DB and queries the corresponding rpf group and ipmc group
+- Associate the corresponding rpf group and ipmc group IDs, and invoke the SAI API to create IP multicast routes.
+
+#### mgmanager
+
+Follow the nexthop group design, and manage the Group and Members separately:

```

Comment:
We don't generally follow manager naming convention to handle routes from ASIC, can we rename it to mroutegrouporch?